accepted/2020/mono-convergence/globalization-support-in-.net5.md

```diff
@@ -0,0 +1,90 @@
+
+### Android
+Android has historically relied on ICU for some kind of globalization support.  From basically the beginning, `Locale` , `Character`, and some classes in `java.text` were provided in the platform globalization.  If you needed more than that, you could access `libicuuc` directly or choose to bundle another version in your APK.  That was pretty much the story up until API level 23.
+
+In API level 24 and higher, the platform white lists its version of `libicuuc` and also includes a subset of [ICU4J](https://developer.android.com/guide/topics/resources/internationalization#relation) API’s.  It’s important to note that [ICU4C](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/) (accessing `libicuuc` directly) is not part of the Android NDK.  What this means is that OEM’s are not required to follow any type of rules and they could limit / alter the data files as they see fit.  The potential for undefined behavior and additional testing on our part is very real. 
+
+Consult this [issue](https://github.com/android/ndk/issues/548#issuecomment-395561629) for detailed commentary on the challenges of making ICU an NDK API.
+
+### What Does This Mean For Us?
+The good news is that p-invoking from managed code works.  Assuming `libicuuc` continues to be white listed, this should not be a problem.  It would be wise if we explored using the `ICU4J`  API subset that’s exposed in the event they decide to tighten things up.  
+
+To summarize:
+
+* We can p-invoke into the system `libicuuc`.  Globalization for .net5 should work normally
+* Android 4.3 (API level 18) started bundling ICU 50 and our minimum version requirement is satisfied. 
+* OEM’s can potentially make life difficult by altering locales / globalization features
+* We should explore trying to interop with the `ICU4J` API and determine what the implications, if any are.
+
+### iOS/tvOS
+There is no build in support for ICU on apple mobile platforms. For our purposes of having highly compatible globalization mode across all platform we support we need to look into bringing ICU to ios/tvOS as well.
+
+System.Globalization.Invariant mode still need to work without any ICU dependency. How we are actually going to slice the actual data will be defined later based on the experience and existing ICU tooling.
+
+We are as expected not the first framework to hit this problem. Swift framework is dealing with same problems and they seem to prefer a solution which drop ICU dependency but that still does not look like final [decision](https://forums.swift.org/t/icu-usage-in-swift/20473).
+
+### WebAssembly
+The intention is set invariant globalization mode as the default mode for everyone for WebAssembly but at the same time allow a fallback to full globalization support when needed. This will come with additional cost but on the web platform we have little bit more flexibility how to deploy and share the data.
```

Comment:
```suggestion
The intention is to set invariant globalization mode as the default mode for everyone for WebAssembly, but at the same time allow a fallback to full globalization support when needed. This will come with additional cost, but on the web platform we have a little bit more flexibility how to deploy and share the data.
```